src/electron/device-connect-view/components/device-connect-port-entry.tsx

```diff
@@ -41,38 +42,35 @@ export class DeviceConnectPortEntry extends React.Component<DeviceConnectPortEnt
                     maskChar=""
                     mask="99999"
                 />
-                <Button
-                    primary={this.props.needsValidation}
-                    disabled={this.state.isValidateButtonDisabled}
-                    className="button-validate-port"
-                    onClick={this.onValidateClick}
-                >
-                    Validate port number
-                </Button>
+                {this.renderValidationPortButton()}
             </div>
         );
     }
 
-    private onPortTextChanged = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
-        this.setState({ isValidateButtonDisabled: !newValue || newValue === '', port: newValue });
-        this.props.updateStateCallback(DeviceConnectState.Default);
-    };
+    private renderValidationPortButton(): JSX.Element {
+        const props = {
+            disabled: this.isValidateButtonDisabled(),
+            className: 'button-validate-port',
+            onClick: this.onValidateClick,
+        };
 
-    private onValidateClick = async (event: React.MouseEvent<HTMLButtonElement>): Promise<void> => {
-        this.setState({ isValidateButtonDisabled: true });
+        if (this.props.viewState.deviceConnectState !== DeviceConnectState.Connected) {
+            return <PrimaryButton {...props}>Validate port number</PrimaryButton>;
```

Comment:
what is the aria label value difference when you use Primary vs DefaultButton instead of using Button with primary attribute ?